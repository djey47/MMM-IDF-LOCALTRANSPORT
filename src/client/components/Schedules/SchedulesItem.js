/* @flow */

import React from 'react';
import classnames from 'classnames';

import { translate, MessageKeys } from '../../../support/messages';
import { Status, MessageKeys as StatusMessageKeys, TimeModes } from '../../../support/status';
import { toWaitingTime, toHoursMinutes } from '../../support/format';
import { now } from '../../support/date';

import type { ServerScheduleResponse } from '../../../types/Transport';
import type { ModuleConfiguration, StationConfiguration } from '../../../types/Configuration';

// TODO style
// import './Traffic.css';

type PropTypes = {
  data: ServerScheduleResponse,
  stop: StationConfiguration,
  config: ModuleConfiguration,
};

/**
 * @private
 */
const resolveName = (/*firstLine: boolean, */stop: StationConfiguration, messages: Object): string => {
  const { line, label, station } = stop;

  // if (!firstLine) return ' ';

  if (label) return label;

  if (line) return line.toString();

  if (station) return station.toString();

  return translate(MessageKeys.UNAVAILABLE, messages);
};

/**
 * @private
 */
const resolveStatus = (statusCode: ?string, messages: Object, statusMessageKeys: Object): string => {
  if (!statusCode) return '';

  const key = statusMessageKeys[statusCode];
  return key && translate(key, messages) || translate(MessageKeys.UNAVAILABLE, messages);
};

/**
 * A traffic info item
 */
const SchedulesItem = ({ data, stop, config }: PropTypes) => {
  // TODO: comingLastUpdate
  const { schedules } = data;
  const {
    messages,
    maxLettersForDestination,
    maxLettersForTime,
    convertToWaitingTime,
    oldUpdateThreshold,
    oldUpdateOpacity,
    oldThreshold,
    updateInterval, 
  } = config;
  const nowMoment = now();  

  return (
    <li className="SchedulesItem">
      <ul className="SchedulesItem__schedules">
        { schedules.map(schedule => {
          const { status, destination, code, time, timeMode } = schedule;
          const itemClassName = classnames('SchedulesItem__schedule', 'bright', {
            'is-delayed': status === Status.DELAYED,
            'is-deleted': status === Status.DELETED,
            'is-ontime': status === Status.ON_TIME,
          });
          let depInfo;
          if (!time) {
            depInfo = translate(MessageKeys.UNAVAILABLE, messages);
          } else if (convertToWaitingTime) {
            depInfo = toWaitingTime(time, nowMoment, messages);
          } else {
            depInfo = toHoursMinutes(time);
          }
          // TODO set into state
          const comingLastUpdate = now();
          const theoricalSuffix = timeMode === TimeModes.THEORICAL ? translate(MessageKeys.THEORICAL, messages) : '';
          const effectiveThreshold = oldUpdateThreshold ? oldUpdateThreshold : updateInterval * (1 + oldThreshold);
          let opacity = '1';
          if (nowMoment.diff(comingLastUpdate).valueOf() > effectiveThreshold) {
            opacity = oldUpdateOpacity.toString();
          }
          const effectiveCode = code || '';

          return (
            <li className={itemClassName} key={`${effectiveCode}-${time || '0'}`}>
              <span className="SchedulesItem__name">
                {resolveName(stop, messages)}
              </span>
              <span className="SchedulesItem__destination" style={{ opacity }}>
                {destination.substr(0, maxLettersForDestination)}
              </span>
              <span className="SchedulesItem__status">
                {`${effectiveCode} ${resolveStatus(status, messages, StatusMessageKeys)}`.trim()}
              </span>
              <span className="SchedulesItem__departure" style={{ opacity }}>
                {depInfo.concat(theoricalSuffix).substr(0, maxLettersForTime)}
              </span>
            </li>
          );
        })
        }
      </ul>
    </li>
  );
};

export default SchedulesItem;
