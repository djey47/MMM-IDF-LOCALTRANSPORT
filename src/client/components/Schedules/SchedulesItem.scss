/** @define SchedulesItem */

.SchedulesItem {
  display: flex;
}

.SchedulesItem__schedules {
  width: 100%;
}

.SchedulesItem__schedule {
  display: flex;
}

.SchedulesItem__schedule.is-delayed {
  color: orange;
}

.SchedulesItem__schedule.is-deleted {
  color: red;
}

/* .SchedulesItem__schedule.is-ontime {} */

.SchedulesItem__name {
  font-weight: bold;
  width: calc(100% / 4);
  padding: 0 4px;
}

.SchedulesItem__destination {
  width: calc(100% / 4);
  padding: 0 4px;
}

.SchedulesItem__status {
  width: calc(100% / 16);
  padding: 0 4px;
}

.SchedulesItem__departure {
  width: calc(100% * 7 / 16);
  padding: 0 4px;
}
