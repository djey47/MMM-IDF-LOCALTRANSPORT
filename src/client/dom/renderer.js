/* @flow */

import React from 'react';
import moment from 'moment-timezone';
import classnames from 'classnames';
import ReactDOM from 'react-dom';

import type Moment from 'moment';

import Main from '../components/Main/Main';
import { toHoursMinutesSeconds, toWaitingTime, toHoursMinutes } from '../support/format';
import { now } from '../support/date';
import { translate, MessageKeys } from '../../support/messages';
import LegacyApi  from '../../support/api/legacy';
import Transilien  from '../../support/api/transilien';
import {
  Status,
  TimeModes,
  MessageKeys as StatusMessageKeys,
}  from '../../support/status';
import { WRAPPER_ID } from '../../support/configuration';

import type { Data, ComingContext } from '../../types/Application';
import type { ModuleConfiguration, StationConfiguration } from '../../types/Configuration';
import type { Schedule, ServerVelibResponse } from '../../types/Transport';

/** Table cells configuration */
const CELLS_COUNT = 4;
const INDEX_STATUS = 3;

/**
 * @returns HTML for main wrapper
 */
export const renderWrapper = (): any => {
  const wrapper = document.createElement('div');
  wrapper.id = WRAPPER_ID;
  wrapper.className = 'IDFTransportWrapper';
  
  return wrapper;
};

/**
 * @returns module header contents
 */
export const renderHeader = (data: Data, config: ModuleConfiguration): string => {
  const { updateInterval, showLastUpdateTime, lastUpdate, showSecondsToNextUpdate, messages } = config;

  let contents = data.header;
  if (showSecondsToNextUpdate) {
    const timeDifference = lastUpdate ? Math.round((updateInterval - now().valueOf() + moment(lastUpdate).valueOf()) / 1000) : 0;
    const secondUnit = translate(MessageKeys.UNITS_SECONDS, messages);
    if (timeDifference > 0) {
      contents += `, ${translate(MessageKeys.NEXT_UPDATE, messages)} ${timeDifference}${secondUnit}`;
    } else if (timeDifference < 0) {
      // TODO when is it supposed to happen? => add unit test if necessary
      contents += `, ${translate(MessageKeys.REQ_UPDATE, messages)} ${Math.abs(timeDifference)}${secondUnit} ${translate(MessageKeys.AGO, messages)}`;
    }
  }

  if (showLastUpdateTime) {
    contents += (lastUpdate ? ` @ ${toHoursMinutesSeconds(lastUpdate)}` : '');
  }

  return contents;
};

/**
 * REACT gateway helper
 */
export const renderMainComponent = (config: ModuleConfiguration, newData?: Object, dataKind?: string): void => {
  ReactDOM.render(
    <Main config={config} newData={newData} dataKind={dataKind} />,
    document.getElementById(WRAPPER_ID)
  );
};

/**
 * @private
 */
const resolveStatus = (statusCode: ?string, messages: Object, statusMessageKeys: Object): string => {
  if (!statusCode) return '';

  const key = statusMessageKeys[statusCode];
  return key && translate(key, messages) || translate(MessageKeys.UNAVAILABLE, messages);
};

/**
 * @private
 */
const resolveName = (firstLine: boolean, stop: StationConfiguration, messages: Object): string => {
  const { line, label, station } = stop;

  if (!firstLine) return ' ';

  if (label) return label;

  if (line) return line.toString();

  if (station) return station.toString();

  return translate(MessageKeys.UNAVAILABLE, messages);
};

/**
 * @private
 */
const renderComingTransport = (firstLine: boolean, stop: StationConfiguration, comingTransport: Schedule, comingLastUpdate: Moment, previous: ComingContext, config: ModuleConfiguration): ?any => {
  const { messages, concatenateArrivals, convertToWaitingTime, maxLettersForDestination, maxLettersForTime, oldUpdateThreshold, updateInterval, oldThreshold, oldUpdateOpacity } = config;
  const nowMoment = now();

  const row = document.createElement('tr');

  const nameCell = document.createElement('td');
  nameCell.className = classnames('align-right');
  nameCell.innerHTML = resolveName(firstLine, stop, messages);
  row.appendChild(nameCell);

  if(!comingTransport) return row;

  const {
    status,
    destination,
    time,
    timeMode,
    code,
  } = comingTransport;

  row.className = classnames('Schedules__item', 'bright', {
    'is-delayed': status === Status.DELAYED,
    'is-deleted': status === Status.DELETED,
    'is-ontime': status === Status.ON_TIME,
  });

  const destinationCell = document.createElement('td');
  destinationCell.innerHTML = destination.substr(0, maxLettersForDestination);
  destinationCell.className = 'align-left';
  row.appendChild(destinationCell);

  const statCell = document.createElement('td');
  statCell.innerHTML = `${code || ''} ${resolveStatus(status, messages, StatusMessageKeys)}`.trim();
  row.appendChild(statCell);

  const depCell = document.createElement('td');
  depCell.className = '';  
  let depInfo;
  if (!time) {
    depInfo = translate(MessageKeys.UNAVAILABLE, messages);
  } else if (convertToWaitingTime) {
    depInfo = toWaitingTime(time, nowMoment, messages);
  } else {
    depInfo = toHoursMinutes(time);
  }
  const theoricalSuffix = timeMode === TimeModes.THEORICAL ? translate(MessageKeys.THEORICAL, messages) : '';
  depCell.innerHTML = depInfo.concat(theoricalSuffix).substr(0, maxLettersForTime);
  row.appendChild(depCell);

  const effectiveThreshold = oldUpdateThreshold ? oldUpdateThreshold : updateInterval * (1 + oldThreshold);
  if (nowMoment.subtract(comingLastUpdate).valueOf() > effectiveThreshold) {
    destinationCell.style.opacity = depCell.style.opacity = oldUpdateOpacity.toString();
  }

  const { previousDestination, previousRow } = previous;
  if (concatenateArrivals 
      && !firstLine 
      && destination === previousDestination) {
    previous.previousDepInfo += ` / ${depInfo}`;
    if (previousRow) {
      previousRow.getElementsByTagName('td')[INDEX_STATUS].innerHTML = previous.previousDepInfo;
    }
    return null;
  } else {
    previous.previousRow = row;
    previous.previousDepInfo = depInfo;
    previous.previousDestination = destination;
    return row;
  }
};

/**
 * @returns HTML for public transport items (rows) for any API
 */
export const renderPublicTransport = (stop: StationConfiguration, stopIndex: ?string, schedules: Object, lastUpdate: Object, config: ModuleConfiguration) => {
  const { maximumEntries, messages } = config;
  const rows = [];
  const unavailableLabel = translate(MessageKeys.UNAVAILABLE, messages);
  const coming: Array<Schedule> = schedules[stopIndex] || [ {
    message: unavailableLabel,
    destination: unavailableLabel,
    code: null,
    status: null,
    timeMode: TimeModes.UNDEFINED,
    time: null,
  } ];
  const comingLastUpdate: Moment = lastUpdate[stopIndex];
  const previous: ComingContext = {
    previousRow: null,
    previousDestination: null,
    previousDepInfo: '',
  };
  let firstLine = true;
  for (let comingIndex = 0; (comingIndex < maximumEntries) && (comingIndex < coming.length); comingIndex++) {
    const row = renderComingTransport(firstLine, stop, coming[comingIndex], comingLastUpdate, previous, config);
    if (row) rows.push(row);
    firstLine = false;
  }
  return rows;
};

/**
 * @returns HTML for public transport items (rows) via classical API (Grimaud v3)
 */
export const renderPublicTransportLegacy = (stop: StationConfiguration, schedules: Object, lastUpdate: Object, config: ModuleConfiguration): any[] => {
  const stopIndex = LegacyApi.createIndexFromStopConfig(stop);

  return renderPublicTransport(stop, stopIndex, schedules, lastUpdate, config);
};

/**
 * @returns HTML for public transport items (rows) via Transilien API
 */
export const renderPublicTransportTransilien = (stop: StationConfiguration, schedules: Object, lastUpdate: Object, config: ModuleConfiguration): any[] => {
  const stopIndex = Transilien.createIndexFromStopConfig(stop);

  return renderPublicTransport(stop, stopIndex, schedules, lastUpdate, config);  
};

/**
 * @private
 */
export const renderNoInfoVelib = (stop: StationConfiguration, messages?: Object): any => {
  const { label, station } = stop;
  const row = document.createElement('tr');
  
  const messageCell = document.createElement('td');
  messageCell.className = 'bright';
  messageCell.innerHTML = `${label || station || MessageKeys.UNAVAILABLE} ${translate(MessageKeys.NOT_YET, messages)}`;
  row.appendChild(messageCell);

  return row;
};

/**
 * @private
 * @returns HTML for info received for Velib (without trend)
 */
export const renderSimpleInfoVelib = (stop: StationConfiguration, station: ServerVelibResponse, messages: Object): any => {
  const row = document.createElement('tr');
  const { label } = stop;
  const { total, bike, empty, name } = station;
  const velibStationCell = document.createElement('td');
  velibStationCell.className = 'align-left';
  velibStationCell.innerHTML = total.toString();
  row.appendChild(velibStationCell);

  const velibStatusCell = document.createElement('td');
  velibStatusCell.className = 'bright';
  velibStatusCell.innerHTML = `${bike} ${translate(MessageKeys.VELIB_BIKES, messages)} ${empty} ${translate(MessageKeys.VELIB_SPACES, messages)}`;
  row.appendChild(velibStatusCell);

  const velibNameCell = document.createElement('td');
  velibNameCell.className = 'align-right';
  velibNameCell.innerHTML = label || name;
  row.appendChild(velibNameCell);

  const velibEmptyCell = document.createElement('td');
  row.appendChild(velibEmptyCell);

  return row;
};

/**
 * @private
 * @returns HTML for info received for Velib (with trend)
 */
export const renderTrendInfoVelib = (stop: StationConfiguration, station: ServerVelibResponse, velibHistory: Object, config: ModuleConfiguration): any => {
  const { name, bike, empty } = station;
  const { velibTrendWidth, velibTrendHeight, velibTrendTimeScale, velibTrendZoom, velibTrendDay } = config;
  const rowTrend = document.createElement('tr');
  const cellTrend = document.createElement('td');

  const trendGraph = document.createElement('canvas');
  trendGraph.className = 'velibTrendGraph';
  trendGraph.width  = velibTrendWidth;
  trendGraph.height = velibTrendHeight;

  const timeScale = velibTrendDay ? 24 * 60 * 60 : velibTrendTimeScale;
  // $FlowFixMe
  trendGraph.timeScale = timeScale;

  const ctx = trendGraph.getContext('2d');
  if (!ctx) { return rowTrend; }

  const nowMoment = now();
  const { label } = stop;
  const currentStation: Array<ServerVelibResponse> = velibHistory[stop.station];
  const { height, width } = trendGraph;
  let previousX = width;
  let inTime = false;
  for (var dataIndex = currentStation.length - 1; dataIndex >= 0 ; dataIndex--) { //start from most recent
    const { lastUpdate, bike, total } = currentStation[dataIndex];
    let dataTimeStamp = nowMoment.subtract(moment(lastUpdate)).seconds(); // time of the event in seconds ago
    if (dataTimeStamp < timeScale || inTime) {
      inTime = dataTimeStamp < timeScale; // compute the last one outside of the time window
      if (dataTimeStamp - timeScale < 10 * 60) { //takes it only if it is within 10 minutes of the closing windows
        dataTimeStamp = Math.min(dataTimeStamp, timeScale); //to be sure it does not exit the graph
        let x, y;
        if (velibTrendDay) {
          if ( dataTimeStamp  < velibTrendZoom ) { //1st third in zoom mode
            x = (1 - dataTimeStamp / velibTrendZoom / 3) * width;
          } else if (dataTimeStamp < timeScale - velibTrendZoom) { //middle in compressed mode
            x = (2 / 3 - (dataTimeStamp - velibTrendZoom) / (timeScale - 2 * velibTrendZoom)/ 3) * width;
          } else {
            x = (1 / 3 - (dataTimeStamp - timeScale + velibTrendZoom)/ velibTrendZoom / 3) * width;
          }
        } else {
          x = (1 - dataTimeStamp / timeScale) * width;
        }
        y = bike / total * height * 4 / 5;
        ctx.fillStyle = 'white';
        ctx.fillRect(x, height - y - 1, previousX - x, Math.max(y, 1)); //a thin line even if it's zero
        previousX = x;
      }
    }
  }
  ctx.font = `${Math.round(height / 5)}px ${ctx.font.split(' ').slice(-1)[0]}`;
  ctx.fillStyle = 'grey';
  ctx.textAlign = 'center';
  ctx.fillText(label || name, width / 2, Math.round(height / 5));
  ctx.textAlign = 'left';
  ctx.fillText(bike.toString(), 10, height - 10);
  ctx.fillText(empty.toString(), 10, Math.round(height / 5) + 10);
  if (velibTrendDay) {
    const text = `${Math.round(velibTrendZoom / 60)}${translate(MessageKeys.UNITS_MINUTES)}`;
    ctx.font = Math.round(height / 10) + 'px ' + ctx.font.split(' ').slice(-1)[0];
    ctx.fillText(text, width * 5 / 6, height / 2);
    ctx.fillText(text, width / 6, height / 2);
    ctx.strokeStyle = 'grey';
    ctx.setLineDash([5, 15]);
    ctx.beginPath();
    ctx.moveTo(2/3 * width, 0);
    ctx.lineTo(2/3 * width, 100);
    ctx.stroke();
    ctx.moveTo(width / 3, 0);
    ctx.lineTo(width / 3, 100);
    ctx.stroke();
    
    // TODO compute properly (use moment)
    // var hourMark = new Date(); var alpha;
    // hourMark.setMinutes(0); hourMark.setSeconds(0);
    // alpha = (hourMark - nowMoment + 24 * 60 * 60 * 1000 - effectiveZoom * 1000) / (24 * 60 * 60 * 1000 - 2 * effectiveZoom * 1000);
    // alpha = (hourMark - nowMoment + effectiveZoom * 1000) / (24 * 60 * 60 * 1000) * width;
    
    // for (var h = 0; h < 24; h = h + 2) {
    //   ctx.fillStyle = 'red';
    //   ctx.textAlign = 'center';
    //   ctx.font = Math.round(height / 12) + 'px';
    //   ctx.fillText(`${(hourMark.getHours() + 24 - h) % 24}`, (2 - h / 24) * width / 3 + alpha, h % 12 * height / 12 / 3 + height / 3);
    // }
  }

  cellTrend.colSpan = CELLS_COUNT; //so that it takes the whole row
  cellTrend.appendChild(trendGraph);
  rowTrend.appendChild(cellTrend);

  return rowTrend;
};

/**
 * @returns HTML for info received for Velib
 */
export const renderVelib = (stop: StationConfiguration, velibHistory: Object, config: ModuleConfiguration): any => {
  const { messages, trendGraphOff } = config;
  const velibStationHistory: Array<ServerVelibResponse> = velibHistory[stop.station];

  if (!velibStationHistory) {
    return renderNoInfoVelib(stop, messages);
  }

  const stationInfo = velibStationHistory.slice(-1)[0];
  if (trendGraphOff) {
    return renderSimpleInfoVelib(stop, stationInfo, messages);
  }

  return renderTrendInfoVelib(stop, stationInfo, velibHistory, config);
};
