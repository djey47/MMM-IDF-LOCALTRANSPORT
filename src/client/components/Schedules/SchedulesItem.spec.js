import React from 'react';
import renderer from 'react-test-renderer';
import moment from 'moment-timezone';
import SchedulesItem from './SchedulesItem';

const mockNow = jest.fn();
jest.mock('../../support/date', () => ({
  now: () => mockNow(),
}));

beforeAll(() => {
  moment.tz.setDefault('UTC');
  mockNow.mockImplementation(() => moment('2017-07-27T08:23:55Z'));
});

describe('SchedulesItem component', () => {
  const baseConfig = {
    convertToWaitingTime: false,
    maxLettersForDestination: 256,
    maximumEntries: 3,
    messages: {},
  };

  const stop = {
    line: ['BUS', 275],
    station: 'Ulbach',
    destination: 'La+Defense',
  };

  it('should render correctly without schedule', () => {
    // given
    const data = {
      schedules: [],
    };

    // when
    const component = renderer.create(
      <SchedulesItem data={data} stop={{}} config={baseConfig} />,
    );

    // then
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render correctly with provided schedules', () => {
    // given
    const data = {
      schedules: [{
        time: '2017-07-16T13:00:00.000Z',
        destination: 'La Défense',
      },{
        time: '2017-07-16T13:05:00.000Z',
        destination: 'Place Charras',       
      }],
    };

    // when
    const component = renderer.create(
      <SchedulesItem data={data} stop={stop} config={baseConfig} />,
    );

    // then
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render correctly with schedule and status info', () => {
    // given
    const rerStop = {
      line: 'A',
      station: 'La+Defense',
      destination: 'a',
    };
    const data = {
      schedules: [{
        time: '2017-05-30T13:00:00.000Z',
        status: 'DELAYED',
        destination: 'La Défense',
        code: 'UAPY',
      }],
    };

    // when
    const component = renderer.create(
      <SchedulesItem data={data} stop={rerStop} config={baseConfig} />,
    );

    // then
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });  

  it('should render correctly with schedule and mission code', () => {
    // given
    const transilienStop = {
      line: 'L',
      station: 'Becon',
    };    
    const data = {
      schedules: [{
        time: '2017-05-30T13:00:00.000Z',
        status: '',
        code: 'POPU',
        destination: 'Saint-Lazare',
      }],
    };

    // when
    const component = renderer.create(
      <SchedulesItem data={data} stop={transilienStop} config={baseConfig} />,
    );

    // then
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();   
  });  

  it('should render correctly with schedule and theorical time mode', () => {
    // given
    const data = {
      schedules: [{
        time: '2017-05-30T13:00:00.000Z',
        timeMode: 'THEORICAL',
        destination: 'La Défense',
      }],
    };

    // when
    const component = renderer.create(
      <SchedulesItem data={data} stop={stop} config={baseConfig} />,
    );

    // then
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();    
  });

  it('should render correctly with schedule and convert to waiting time', () => {
    // given
    mockNow.mockImplementation(() => moment('2017-05-30T12:45:00.000Z'));
    const data = {
      schedules: [{
        time: '2017-05-30T13:00:00.000Z',
        destination: 'La Défense',
      },{
        time: '2017-05-30T13:15:00.000Z',
        destination: 'Place Charras',       
      }],
    };
    const config = {
      ...baseConfig,
      convertToWaitingTime: true,
    };

    // when
    const component = renderer.create(
      <SchedulesItem data={data} stop={stop} config={config} />,
    );

    // then
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();        
  });  
});
