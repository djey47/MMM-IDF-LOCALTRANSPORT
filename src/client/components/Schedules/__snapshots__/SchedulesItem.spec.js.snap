// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SchedulesItem component should render correctly with provided schedules 1`] = `
<li
  className="SchedulesItem"
>
  <ul
    className="SchedulesItem__schedules"
  >
    <li
      className="SchedulesItem__schedule bright"
    >
      <span
        className="SchedulesItem__name"
      >
        BUS,275
      </span>
      <span
        className="SchedulesItem__destination"
        style={
          Object {
            "opacity": "1",
          }
        }
      >
        La Défense
      </span>
      <span
        className="SchedulesItem__status"
      >
        
      </span>
      <span
        className="SchedulesItem__departure"
        style={
          Object {
            "opacity": "1",
          }
        }
      >
        13:00
      </span>
    </li>
    <li
      className="SchedulesItem__schedule bright"
    >
      <span
        className="SchedulesItem__name"
      >
        BUS,275
      </span>
      <span
        className="SchedulesItem__destination"
        style={
          Object {
            "opacity": "1",
          }
        }
      >
        Place Charras
      </span>
      <span
        className="SchedulesItem__status"
      >
        
      </span>
      <span
        className="SchedulesItem__departure"
        style={
          Object {
            "opacity": "1",
          }
        }
      >
        13:05
      </span>
    </li>
  </ul>
</li>
`;

exports[`SchedulesItem component should render correctly with schedule and convert to waiting time 1`] = `
<li
  className="SchedulesItem"
>
  <ul
    className="SchedulesItem__schedules"
  >
    <li
      className="SchedulesItem__schedule bright"
    >
      <span
        className="SchedulesItem__name"
      >
        BUS,275
      </span>
      <span
        className="SchedulesItem__destination"
        style={
          Object {
            "opacity": "1",
          }
        }
      >
        La Défense
      </span>
      <span
        className="SchedulesItem__status"
      >
        
      </span>
      <span
        className="SchedulesItem__departure"
        style={
          Object {
            "opacity": "1",
          }
        }
      >
        15 {units.minutes}
      </span>
    </li>
    <li
      className="SchedulesItem__schedule bright"
    >
      <span
        className="SchedulesItem__name"
      >
        BUS,275
      </span>
      <span
        className="SchedulesItem__destination"
        style={
          Object {
            "opacity": "1",
          }
        }
      >
        Place Charras
      </span>
      <span
        className="SchedulesItem__status"
      >
        
      </span>
      <span
        className="SchedulesItem__departure"
        style={
          Object {
            "opacity": "1",
          }
        }
      >
        30 {units.minutes}
      </span>
    </li>
  </ul>
</li>
`;

exports[`SchedulesItem component should render correctly with schedule and mission code 1`] = `
<li
  className="SchedulesItem"
>
  <ul
    className="SchedulesItem__schedules"
  >
    <li
      className="SchedulesItem__schedule bright"
    >
      <span
        className="SchedulesItem__name"
      >
        L
      </span>
      <span
        className="SchedulesItem__destination"
        style={
          Object {
            "opacity": "1",
          }
        }
      >
        Saint-Lazare
      </span>
      <span
        className="SchedulesItem__status"
      >
        POPU
      </span>
      <span
        className="SchedulesItem__departure"
        style={
          Object {
            "opacity": "1",
          }
        }
      >
        13:00
      </span>
    </li>
  </ul>
</li>
`;

exports[`SchedulesItem component should render correctly with schedule and status info 1`] = `
<li
  className="SchedulesItem"
>
  <ul
    className="SchedulesItem__schedules"
  >
    <li
      className="SchedulesItem__schedule bright is-delayed"
    >
      <span
        className="SchedulesItem__name"
      >
        A
      </span>
      <span
        className="SchedulesItem__destination"
        style={
          Object {
            "opacity": "1",
          }
        }
      >
        La Défense
      </span>
      <span
        className="SchedulesItem__status"
      >
        UAPY {status.delayed}
      </span>
      <span
        className="SchedulesItem__departure"
        style={
          Object {
            "opacity": "1",
          }
        }
      >
        13:00
      </span>
    </li>
  </ul>
</li>
`;

exports[`SchedulesItem component should render correctly with schedule and theorical time mode 1`] = `
<li
  className="SchedulesItem"
>
  <ul
    className="SchedulesItem__schedules"
  >
    <li
      className="SchedulesItem__schedule bright"
    >
      <span
        className="SchedulesItem__name"
      >
        BUS,275
      </span>
      <span
        className="SchedulesItem__destination"
        style={
          Object {
            "opacity": "1",
          }
        }
      >
        La Défense
      </span>
      <span
        className="SchedulesItem__status"
      >
        
      </span>
      <span
        className="SchedulesItem__departure"
        style={
          Object {
            "opacity": "1",
          }
        }
      >
        13:00{theorical}
      </span>
    </li>
  </ul>
</li>
`;

exports[`SchedulesItem component should render correctly without schedule 1`] = `
<li
  className="SchedulesItem"
>
  <ul
    className="SchedulesItem__schedules"
  />
</li>
`;
